jazz_services-handler/index.js

```diff
@@ -22,605 +22,283 @@ This Handler looks for Service Creation events and updates Service Catalog
-							"processed_events": processedEvents.length,
-							"failed_events": failedEvents.length
-						});
-					}
-					
-					
-				});
-			}catch(e){
-				mainCallback(e);
-			}
+var validateAuthToken = function (result) {
+	return new Promise((resolve, reject) => {
+		if (result.statusCode === 200 && result.body && result.body.data) {
+			return resolve(result.body.data.token);
+		} else {
+			return reject(errorHandler.throwUnauthorizedError("User is not authorized to access this service"));
 		}
-	}, function (err, results) {
-		logger.error("### Error : " + JSON.stringify(err));
-		if (err) {
-			cb(err);
+	});
+}
+
+var handleError = function (errorType, message) {
+	var error = {};
+	error.failure_code = errorType;
+	error.failure_message = message;
+	return error;
+}
+
+var processRecords = function (event, configData, authToken) {
+	return new Promise((resolve, reject) => {
+		var processRecordPromises = [];
+		for (i = 0; i < event.Records.length; i++) {
+			processRecordPromises.push(processRecord(event.Records[i], configData, authToken));
+		}
+		Promise.all(processRecordPromises)
+			.then((result) => { return resolve(result); })
+			.catch((error) => { return reject(error); });
+	});
+}
+
+var processRecord = function (record, configData,authToken) {
+	return new Promise((resolve, reject) => {
+		var sequenceNumber = record.kinesis.sequenceNumber;
+		var encodedPayload = record.kinesis.data;
+		var payload;
+		return checkInterest(encodedPayload, sequenceNumber)
+			.then(result => {
+				payload = result.payload;
+				if (result.interested_event) {
+					return processEvent(payload, configData, authToken);
+				} else {
+					handleProcessedEvents(sequenceNumber, payload);
+				}
+			})
+			.then(result => {
+				handleProcessedEvents(sequenceNumber, payload);
+				return resolve(result);
+			})
+			.catch(err => {
+				handleFailedEvents(sequenceNumber, err.failure_message, payload, err.failure_code);
+				return reject(err);
+			});
+	});
+}
+var checkInterest = function (encodedPayload, sequenceNumber) {
+	return new Promise((resolve, reject) => {
+		var payload = JSON.parse(new Buffer(encodedPayload, 'base64').toString('ascii'));
+		if (payload.Item.EVENT_TYPE && payload.Item.EVENT_TYPE.S) {
+			logger.info("found " + payload.Item.EVENT_TYPE.S + " event with sequence number: " + sequenceNumber);
+			resolve({
+				"interested_event": true,
+				"payload": payload.Item
+			});
 		} else {
-			cb(null, results);
+			resolve({
+				"interested_event": false,
+				"payload": payload.Item
+			});
 		}
 	});
+}
+
+var processEvent = function (payload, configData, authToken) {
+	return new Promise((resolve, reject) => {
+		if (!payload.EVENT_NAME.S || !payload.EVENT_STATUS.S) {
+			logger.error("validation error. Either event name or event status is not properly defined.");
+			var err = handleError(failureCodes.PR_ERROR_1.code, "Validation error while processing event for service");
+			return reject(err);
```

Comment:
Don't reject, just resolve.

Reply:
This wasn't fixed and code was merged. Please create another PR with the changes requested in this PR.